From: Didier Raboud <odyx@debian.org>
Date: Tue, 9 Aug 2016 18:11:49 +0200
Subject: Debian: Use dpkg-architecture in cups-config to make it
 architecture-independant

 Also filter -L/usr/lib/(triplet) out of LDFLAGS.
 Also strip unnecessary libs from --libs.
 Also launch krb5-config at runtime to avoid getting usr/lib/(triplet) from krb5-config

Bug-Debian: https://bugs.debian.org/741519
Bug-Debian: https://bugs.debian.org/727058
Bug-Debian: https://bugs.debian.org/730838
Bug-Debian: https://bugs.debian.org/751157
Forwarded: not-needed
---
 cups-config.in | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/cups-config.in b/cups-config.in
index d081df9..efe2bf9 100755
--- a/cups-config.in
+++ b/cups-config.in
@@ -18,7 +18,7 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 bindir=@bindir@
 includedir=@includedir@
-libdir=@libdir@
+libdir=${prefix}/lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH)
 datarootdir=@datadir@
 datadir=@datadir@
 sysconfdir=@sysconfdir@
@@ -30,7 +30,8 @@ INSTALLSTATIC=@INSTALLSTATIC@
 # flags for compiler and linker...
 CFLAGS=""
 LDFLAGS="@EXPORT_LDFLAGS@"
-LIBS="@LIBGSSAPI@ @DNSSDLIBS@ @EXPORT_TLSLIBS@ @LIBZ@ @LIBS@"
+LIBGSSAPI=`[ ! -x /usr/bin/krb5-config ] || /usr/bin/krb5-config --libs`
+LIBS="$LIBGSSAPI @DNSSDLIBS@ @EXPORT_SSLLIBS@ @LIBZ@ @LIBS@"
 
 # Check for local invocation...
 selfdir=`dirname $0`
@@ -44,7 +45,7 @@ else
 	CFLAGS="$CFLAGS -I$includedir"
     fi
 
-    if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib64; then
+    if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib64 -a $libdir != /usr/lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH); then
 	LDFLAGS="$LDFLAGS -L$libdir"
     fi
 fi
